### PR TITLE
Add most read on front pages

### LINKED ISSUE #5432: Integrate MostRead into Front Pages and Article Pages
**Is your feature request related to a problem? Please describe.**
Now that the container has been merged into Simorgh, we can put MostRead in front pages and article pages. There are various differences between MostRead in front pages and articles such as the landmarks (inside the main in front pages, outside the main in article pages). This also will combine issues https://github.com/bbc/simorgh/issues/5370 and https://github.com/bbc/simorgh/issues/5237. 

Describe the solution you'd like
**To be completed in this issue:**
- Render `MostReadContainer` in `FrontPageMain` and `ArticleMain`.
- Add check to see which page type Most Read is being rendered on.
- Add check to see if the page has a `UsefulLinks` block anywhere on the page. This can be found in `blockTypes` in Ares feeds. Can also search for `semanticGroupName "Useful links"`. If so, render MostRead between `featured-site-top-stories` and `useful-links`.

**For front pages:**
- Ensure that the component is rendered within the main landmark.
- To be confirmed by UX:
   - Make section label have a grey line like other sections on the page.
   - Confine the component with styles from `gelWidthMax` in `./src/app/components/grid`

**For article pages:**
- Ensure that the component is rendered outside the main landmark.
- To be confirmed by UX:
   - Confine the component with styles from `gelWidthMax` in `./src/app/components/grid`

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by thekp**
Resolves #5432

**Overall change:** 
_Integrate Most read into front pages for all services (on local/test environment)_

**Code changes:**
- _Added Most read container to front page container_
- _Toggle most read on local and test environment_
- _updated local route for most read data to be `./mostread.json` as live data endpoint is: https://www.bbc.com/mundo/mostread.json_
- _added some functionality to conditionally render most read above useful links _
- _add `F_REITH_SERIF_LIGHT` to services using REITH SERIF, as it is required for the numeral font_
---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

### CODE REVIEW

**Review by j-pendlebury — APPROVED** [review]

File: src/app/containers/MostRead/Canonical/index.jsx
Comment (tochwill) [inline src/app/containers/MostRead/Canonical/index.jsx]: Do you need the max-width in here?

Comment (mmcculloch) [reply src/app/containers/MostRead/Canonical/index.jsx]: The below media query `@media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {` applies different styles after this max so it just avoids them overriding. We're putting this one on hold till tomorrow so we can Jim some questions. We might not need Group 5 in the new layout.

**Comment by thekp** [comment]
This PR is affecting the front page cypress tests here: https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/simorgh/detail/addMostReadOnPages/19/pipeline

I'm gonna look into it 👀

File: src/app/containers/FrontPageMain/index.jsx
Comment (OlgaLyubin) [inline src/app/containers/FrontPageMain/index.jsx]: Should this not be `useful-links` with a hyphen like in Ares data?

Comment (thekp) [reply src/app/containers/FrontPageMain/index.jsx]: yep, good spot.

File: src/app/containers/FrontPageMain/index.jsx
Comment (tochwill) [inline src/app/containers/FrontPageMain/index.jsx]: Can you look at the group type instead like you do in the function above?

File: src/app/containers/MostRead/Canonical/index.jsx
Comment (tochwill) [inline src/app/containers/MostRead/Canonical/index.jsx]: Do you need this fragment?

File: src/app/containers/MostRead/index.jsx
Comment (tochwill) [inline src/app/containers/MostRead/index.jsx:13]: Is there any logic to only render the Canonical component on canonical pages?

File: src/app/containers/MostRead/Canonical/index.jsx
Comment (thekp) [reply src/app/containers/MostRead/Canonical/index.jsx]: I just tried to remove it, it seems like I need the fragment there in order to write the JS logic within JSX.

image

Comment (thekp) [reply src/app/containers/MostRead/Canonical/index.jsx]: had to remove curly braces, cheers Tom

File: src/app/containers/MostRead/index.jsx
Comment (thekp) [reply src/app/containers/MostRead/index.jsx:13]: not atm 👀

Comment (thekp) [reply src/app/containers/MostRead/index.jsx:13]: just added in some logic now, to enable most read when it's not AMP 🤞

**Review by DenisHdz — APPROVED** [review]
Great job 👍

**Comment by thekp** [comment]
At the largest viewport, most read is using a 5 column view, however we do not want this and this will be addressed in a future psamemad issue:https://github.com/bbc/psammead/issues/3108
image


a side from this most read works on all over viewports as expected: 
image
image

File: src/app/containers/FrontPageMain/index.jsx
Comment (tochwill) [inline src/app/containers/FrontPageMain/index.jsx:98]: Just to be 100% sure, we can only have 1 useful links group on the page?

**Review by j-pendlebury — APPROVED** [review]

File: src/app/containers/MostRead/Canonical/index.jsx
Comment (DenisHdz) [inline src/app/containers/MostRead/Canonical/index.jsx]: I believe we should be adding some spacing between the `Section Label` and the `MostReadList`, to keep it consistent with the rest of sections https://github.com/bbc/simorgh/blob/latest/src/app/containers/FrontPageSection/index.jsx#L115

File: src/app/containers/FrontPageMain/index.jsx
Comment (thekp) [reply src/app/containers/FrontPageMain/index.jsx:98]: I just confirmed with @jimjohnsonrollings irl, and he said "yes there should only ever be one useful links or none"

File: src/app/containers/MostRead/Canonical/index.jsx
Comment (thekp) [reply src/app/containers/MostRead/Canonical/index.jsx]: good spot, will add this now 🙇

File: src/app/containers/MostRead/Canonical/LastUpdated.jsx
Comment (mmcculloch) [inline src/app/containers/MostRead/Canonical/LastUpdated.jsx]: Space here we can remove

File: src/app/containers/MostRead/Canonical/index.jsx
Comment (mmcculloch) [inline src/app/containers/MostRead/Canonical/index.jsx]: Space here isn't needed

Comment (thekp) [reply src/app/containers/MostRead/Canonical/index.jsx]: will remove now 👍

File: src/app/containers/MostRead/Canonical/index.jsx
Comment (mmcculloch) [inline src/app/containers/MostRead/Canonical/index.jsx]: Getting a warning here that `timezone` should be included

Comment (thekp) [reply src/app/containers/MostRead/Canonical/index.jsx]: ah yes, forgot to add timezone here previously 👍

**Comment by PriyaKR** [comment]
@thekp do we have a non js fallback for the most read component?

**Comment by thekp** [comment]
> @thekp do we have a non js fallback for the most read component?

nope, we don't expect this work with noJS as it depends on a `fetch` call

**Comment by PriyaKR** [comment]
Testing:
On canonical front pages
- Looks good across browsers and devices no js fallback which is expected.Would be good to verify the typography for the headline across different viewports with UX.
- The timestamp is not rendered if the article was published today which is as expected.
- Quick accessibility testing was done found the similar issues which we found in the ally swarm https://docs.google.com/spreadsheets/d/1fL1IaWHfgjVnx3bt_ux2YN4mRR3pya9NDKHh15Vp0d8/edit#gid=0
- Verified this case as well https://github.com/bbc/simorgh/issues/3534
- When the article is less than 60 days old the timestamp doesn't get rendered which is as expected.

Note:Above 1008px the most read appear in 5 columns which is not expected have a separate PR raised.

**Comment by thekp** [comment]
We will do the UX review once, we've merged this PR as well: https://github.com/bbc/psammead/pull/3116 and this issue: https://github.com/bbc/psammead/issues/3122

**Pull request merged**
